test/fixtures/change-code.js

```diff
@@ -77,7 +79,7 @@ function listener (code, signal) {
 function run (opt) {
   console.error(opt)
   var shell = process.platform === 'win32' ? null : { shell: '/bin/bash' }
-  exec(process.execPath + ' ' + __filename + ' ' + opt, shell, function (err, stdout, stderr) {
+  exec(join(process.execPath, ' ', __filename, ' ' + opt), shell, function (err, stdout, stderr) {
```

Comment:
That sounds like a bug in standard.  Especially since you can still throw spaces in there anyway.

Would it be ok with you doing `exec(path.resolve(process.execPath) + ' ' + __filename + ' ' + opt)`?  Ie, just resolve the execPath, not the whole command line.